src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -741,22 +744,30 @@ bool InputStateMachineEngine::_WriteMouseEvent(const size_t column, const size_t
 // - Retrieves the modifier state from a set of parameters for a cursor keys
 //      sequence. This is for Arrow keys, Home, End, etc.
 // Arguments:
-// - rgusParams - the set of parameters to get the modifier state from.
-// - cParams - the number of elements in rgusParams
+// - parameters - the set of parameters to get the modifier state from.
 // Return Value:
 // - the INPUT_RECORD compatible modifier state.
 DWORD InputStateMachineEngine::_GetCursorKeysModifierState(const std::basic_string_view<size_t> parameters) noexcept
 {
-    // Both Cursor keys and generic keys keep their modifiers in the same index.
-    return _GetGenericKeysModifierState(parameters);
+    DWORD modifiers = 0;

```

Comment:
so just so i understand this: _all cursor keys are enhanced_ but only generic keys _<= `Next`_ are enhanced?

I'm worried that the logic here is duplicated, but I also don't mind terribly?

Reply:
It looks like F1, F2, F3 and F4 also go through GetCursorKeysModifierState ... which makes it look like F1-F4 will become `ENHANCED` here...

Reply:
is that intended?

Reply:
There's a _little_ duplication but that's because the vkey for generics is saved in `parameters` whereas for Cursor keys it's in the wch.

I'll remove counting F1-F4 as enhanced now.